boat-maven-plugin/src/main/java/com/backbase/oss/boat/radio/RadioMojo.java

```diff
@@ -0,0 +1,255 @@
+     *     For example - By default {@code my-service-api-v3.1.4.yaml} would be evaluated to {@code my-service-api}
+     * </p>
+     * <p>
+     *     {@code name} :
+     *     Spec Name in Boat-Bay. Defaults to filename.
+     * </p>
+     * <p>
+     *     {@code inputSpec} :
+     *      Location of the OpenAPI spec, as URL or local file glob pattern.
+     *      If the input is a local file, the value of this property is considered a glob pattern that must
+     *      resolve to a unique file.
+     *      The glob pattern allows to express the input specification in a version neutral way. For
+     *      instance, if the actual file is {@code my-service-api-v3.1.4.yaml} the expression could be
+     *      {@code my-service-api-v*.yaml}.
+     * </p>
+     *
+     */
+    @Parameter(property = "specs", required = true)
+    private SpecConfig[] specs;
+
+    /**
+     * Output directory for boat-radio report.
+     */
+    @Parameter(name = "radioOutput", defaultValue = "${project.build.directory}/target/boat-radio-report" )
+    private File radioOutput;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        BasicAuthRequestInterceptor basicAuthRequestInterceptor = null;
+
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        if (username != null && !username.isEmpty() && password != null && !password.isEmpty()) {
+            getLog().info("Basic Authentication set for username "+username);
+            basicAuthRequestInterceptor = new BasicAuthRequestInterceptor(username, password);
+        } else {
+            getLog().info("No Authentication set");
+        }
+
+        List<UploadSpec> allSpecs = new ArrayList<>();
+
+        for (SpecConfig spec : specs) {
+            allSpecs.add(mapToUploadSpec(spec));
+        }
+
+        ApiClient apiClient = new ApiClient().setBasePath(boatBayUrl);
+        if (basicAuthRequestInterceptor != null) {
+            apiClient.addAuthorization("Basic Auth", basicAuthRequestInterceptor);
+        }
+
+        BoatMavenPluginApi api = apiClient.buildClient(BoatMavenPluginApi.class);
+
+        UploadRequestBody uploadRequestBody = new UploadRequestBody();
+        uploadRequestBody.setGroupId(groupId);
+        uploadRequestBody.setArtifactId(artifactId);
+        uploadRequestBody.setVersion(version);
+        uploadRequestBody.setSpecs(allSpecs);
+
+        List<BoatLintReport> reports = api.uploadSpec(portalKey, sourceKey, uploadRequestBody);
+
+        try {
+            File outputFile = new File(getOutput(), "radioOutput.json");
+            objectMapper.writerWithDefaultPrettyPrinter().writeValue(outputFile, reports);
+            reports.forEach(report -> getLog().info(format("Changes to spec %s is %s", report.getSpec().getKey(), report.getSpec().getChanges())));
+            getLog().info("UPLOAD TO BOAT-BAY SUCCESSFUL, check the full report: " + outputFile.getCanonicalPath());
```

Comment:
We should also have boat-bay radio fail the build (configurable) when it's not downwards compatible or there are too many lint violations?

Reply:
Something like?
```
if (failOnWarning) {
                throw new MojoFailureException("Message");
            }
```